File: Ejercicio.WebApp/App/Controller/TiendasController.js
'use strict';
app.controller('TiendasController', ['$scope', 'TiendaServiceFactory', '$state', '$rootScope', function ($scope, TiendaServiceFactory, $state, $rootScope) {
    $scope.stores = [];

    $scope.init = function () {
        $scope.getStores();
    };

    $scope.getStores = function () {
        TiendaServiceFactory.getAllStores()
            .then(function (response) {
                if (response.status === 200) {
                    $scope.stores = response.data;
                }
                else {
                    $rootScope.showToast("Ocurrió un error.");
                }
            })
            .catch(function () {
                $rootScope.showToast("Ocurrió un error.");
            });
    };

    $scope.addStore = function () {
        $state.go('Tienda', { store: null });
    };

    $scope.updateStore = function (storeToUpdate) {
        $state.go('Tienda', { store: storeToUpdate });
    };

    $scope.deleteStore = function (idStore) {
        TiendaServiceFactory.deleteStore(idStore)
            .then(function (response) {
                if (response.status === 200) {
                    $scope.getStores();
                    $rootScope.showToast("Registro eliminado.");
                }
                else {
                    $rootScope.showToast("Ocurrió un error.");
                }
            })
            .catch(function () {
                $rootScope.showToast("Ocurrió un error.");
            });
    };

    $scope.init();
}]);

File: Ejercicio.Api/Controllers/TiendasController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Ejercicio.DataAccess.Models;
using Ejercicio.Domain.Repositories;
using Microsoft.AspNetCore.Mvc;

namespace Ejercicio.Api.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class TiendasController : ControllerBase
    {
        private readonly ITiendaRepository _repository;

        public TiendasController(ITiendaRepository repository)
        {
            _repository = repository;
        }

        [HttpGet]
        public async Task<ActionResult> GetStores()
        {
            try
            {
                List<Tienda> tiendas = await _repository.GetStores();

                if (tiendas == null)
                {
                    return BadRequest();
                }

                return Ok(tiendas);
            }
            catch (Exception)
            {
                throw new ArgumentException("Ocurrió un error en el servidor");
            }
        }

        [HttpPost]
        public async Task<ActionResult> AddStore(Tienda tienda)
        {
            try
            {
                int result = await _repository.AddStore(tienda);

                if (result.Equals(0))
                {
                    return BadRequest();
                }

                return Ok();
            }
            catch (Exception ex)
            {
                throw new ArgumentException("Ocurrió un error en el servidor");
            }
        }

        [HttpPut]
        public async Task<ActionResult> UpdateStore(Tienda tienda)
        {
            try
            {
                int result = await _repository.UpdateStore(tienda);

                if (result.Equals(0))
                {
                    return BadRequest();
                }

                return Ok();
            }
            catch (Exception ex)
            {
                throw new ArgumentException("Ocurrió un error en el servidor");
            }
        }

        [HttpDelete]
        public async Task<ActionResult> DeleteStore(int idStore)
        {
            try
            {
                Tienda store = _repository.GetStores().Result.First(x => x.IdTienda == idStore);

                int result = await _repository.DeleteStore(store);

                if (result.Equals(0))
                {
                    return BadRequest();
                }

                return Ok();
            }
            catch (Exception)
            {
                throw new ArgumentException("Ocurrió un error en el servidor");
            }
        }
    }
}